pkg/bitcoin/address.go

```diff
@@ -97,30 +123,17 @@ func (s *Service) EncodeAddress(
 			}
 
 			return btcutil.NewAddressScriptHash(redeemScript, chainParams)
-		case pb.AddressEncoding_ADDRESS_ENCODING_P2WPKH:
+		case NativeSegwit:
 			// Ref: https://bitcoincore.org/en/segwit_wallet_dev/#native-pay-to-witness-public-key-hash-p2wpkh
 			return btcutil.NewAddressWitnessPubKeyHash(publicKeyHash, chainParams)
 		default:
-			return nil, btcutil.ErrUnknownAddressType
+			return nil, ErrUnknownAddressType
 		}
 	}()
 
 	if err != nil {
-		return nil, status.Errorf(codes.Internal, err.Error())
+		return "", err
```

Comment:
Same: Using [`pkg/errors`](https://github.com/pkg/errors) would allow to wrap and give context like:
```
if err != nil {
        return "", errors.Wrapf(err, "Unable to encode key %s to address", publicKey)
}
```